pkg/webui/console/components/payload-formatters-form/index.js

```diff
@@ -161,67 +225,112 @@ class PayloadFormattersForm extends React.Component {
     }
   }
 
+  @bind
+  _showTestSection() {
+    const { allowTest, defaultType } = this.props
+    const { type } = this.state
+
+    // Show the testing section if:
+    // 1. This payload formatters form is linked to this end device.
+    if (!allowTest) {
+      return false
+    }
+    // 2. This end device is set to use the application level formatter and it is not set to `NONE`.
+    if (type === TYPES.DEFAULT) {
+      return defaultType !== TYPES.NONE
+    }
+    // 3. The end device formatter is not set to `NONE`.
+    return type !== TYPES.NONE
+  }
+
   render() {
     const { initialType, initialParameter, uplink, allowReset } = this.props
-    const { error, type } = this.state
+    const { error, type, test } = this.state
 
     const initialValues = {
-      [FIELD_NAMES.RADIO]: type,
+      [FIELD_NAMES.SELECT]: type,
       [FIELD_NAMES.JAVASCRIPT]:
         initialType === TYPES.JAVASCRIPT ? initialParameter : getDefaultJavascriptFormatter(uplink),
       [FIELD_NAMES.GRPC]:
         initialType === TYPES.GRPC ? initialParameter : getDefaultGrpcServiceFormatter(uplink),
     }
+    const options = allowReset ? formatterOptionsWithReset : formatterOptions
 
     return (
-      <div>
-        <Form
-          submitEnabledWhenInvalid
-          onSubmit={this.handleSubmit}
-          initialValues={initialValues}
-          validationSchema={validationSchema}
-          error={error}
-        >
-          <Form.Field
-            name={FIELD_NAMES.RADIO}
-            title={m.formatterType}
-            component={Radio.Group}
-            onChange={this.onTypeChange}
-            warning={type === TYPES.DEFAULT ? m.appFormatterWarning : undefined}
-            horizontal
+      <Row>
+        <Col sm={12} lg={this._showTestSection() ? 6 : 12}>
```

Comment:
```suggestion
        <Col sm={12} lg={this._showTestSection() ? 6 : 12}>
```

Are we using this underscore consistently?

Reply:
What do you mean?

Reply:
The method uses an underscore prefix. I was just wondering when this should be used.

Reply:
We already use that approach, e.g.
https://github.com/TheThingsNetwork/lorawan-stack/blob/v3.12/pkg/webui/components/key-value-map/entry.js#L31-L41
https://github.com/TheThingsNetwork/lorawan-stack/blob/v3.12/pkg/webui/console/components/join-eui-prefixes-input/index.js#L71-L81

The reason I added prefix to `showTestSection` is to visually separate regular methods that, say, can be passed down to children and strictly utilitarian method that is used only within the component, should not be passed anywhere else and needs access to component props.

Reply:
Consider adding this to the coding guidelines so we can ensure that this is done consistently.